internal/venice-common/src/main/java/com/linkedin/venice/pubsub/api/PubSubSecurityProtocol.java

```diff
@@ -0,0 +1,72 @@
+package com.linkedin.venice.pubsub.api;
+
+import com.linkedin.venice.utils.ReflectUtils;
+import java.util.Locale;
+
+
+/**
+ * This enum is equivalent to Kafka's SecurityProtocol enum.
+ *
+ * We need this abstraction because Kafka's enum is present in two different namespaces, which are different between
+ * LinkedIn's fork and the Apache fork.
+ */
+public enum PubSubSecurityProtocol {
```

Comment:
(Maybe not part of this PR), but in my opinion, the communication protocol should be part of the adapter layer configuration and ideally invisible in the rest of the codebase, except in the adapter config class and where we set it.

Reply:
Hmm... what is the adapter config you speak of? Is this already part of the code base or some future thing?

Reply:
E.g. ApacheKafkaProducerConfig/ApacheKafkaConsumerConfig

So we'll pass config `"kafka.security.protocol": "ssl"` , then inside ApacheKafkaProducerConfig/ApacheKafkaConsumerConfig we will convert this string to `org.apache.kafka.common.protocol.SecurityProtocol.SSL`.

Reply:
What are you proposing concretely? Are you saying that we should not have this enum at all? Currently, we use it in several code paths, mostly to validate the passed in values.

I could envision that if we added support for a very different pub sub (e.g. Pulsar, Kinesis or whatever) then maybe those enum values may not be spelled out exactly the same in that other system, but until we cross that bridge, IDK if it's worth removing this enum from the codebase...

Reply:
> What are you proposing concretely? Are you saying that we should not have this enum at all? Currently, we use it in several code paths, mostly to validate the passed in values.

Yes, I think we should not use this enum at all. Any configurations specific to a pub-sub client should be extracted into that client's configuration class and should not be exposed to other parts of the code. However, since this is out of scope for this PR, we can leave it for now.